File: tests/fixtures/sample_project_java/src/main/java/com/example/db/Connection.java
package com.example.db;

/** Behaviour every backing store must provide. */
interface DataStore {
    /** Run a statement and return the number of affected rows. */
    int execute(String sql);
}

/** A pooled connection to the fixture database. */
public class Connection implements DataStore {
    /** The backing database URL. */
    public String url;
    private boolean open;

    /** Return a process-wide connection to the fixture database. */
    public static Connection getConnection() {
        Connection c = new Connection();
        c.url = "jdbc:sqlite:fixture.db";
        c.open = true;
        return c;
    }

    /** Close the connection and release the pool slot. */
    public void close() {
        this.open = false;
    }

    @Override
    public int execute(String sql) {
        return sql.length();
    }

    /** Create the fixture tables if they don't exist yet. */
    public static boolean ensureSchema(Connection conn) {
        return conn.execute("CREATE TABLE users") > 0;
    }
}

File: tests/fixtures/sample_project_csharp/src/Db/Connection.cs
namespace Acme.Db;

/// <summary>Behaviour every backing store must provide.</summary>
public interface DataStore
{
    /// <summary>Run a statement and return the number of affected rows.</summary>
    int Execute(string sql);
}

/// <summary>A pooled connection to the fixture database.</summary>
public class Connection : DataStore
{
    /// <summary>The backing database URL.</summary>
    public string Url;
    private bool Open;

    /// <summary>Return a process-wide connection to the fixture database.</summary>
    public static Connection GetConnection()
    {
        Connection c = new Connection();
        c.Url = "sqlite:fixture.db";
        c.Open = true;
        return c;
    }

    /// <summary>Close the connection and release the pool slot.</summary>
    public void Close()
    {
        this.Open = false;
    }

    public int Execute(string sql)
    {
        return sql.Length;
    }

    /// <summary>Create the fixture tables if they don't exist yet.</summary>
    public static bool EnsureSchema(Connection conn)
    {
        return conn.Execute("CREATE TABLE users") > 0;
    }
}
